### ISSUE TITLE
Add MapWalk to Share Recipient list?

### ISSUE THREAD

**Issue by tedbarnett** [tedbarnett/MapWalk#25]
When I send myself a KML and it appears in my Messages, there is no way to "open" it.  Is it possible to register MapWalk as an app that can open KML files so that it opens in MapWalk?

**Comment by bhautikpatoliya004** [comment]
@tedbarnett Do you mean to display our app in the share(Action) list?
Please see attached screenshot.
<img width="535" alt="Screenshot 2023-10-04 at 6 17 47 PM" src="https://github.com/tedbarnett/MapWalk/assets/114083934/b6ac96f7-13f5-42a8-9b56-3e0fed7b284c">

**Comment by tedbarnett** [comment]
Yes, if that's possible!  As an app that can open KML files, I assume.

**Comment by bhautikpatoliya004** [comment]
@tedbarnett To run the action extension in the device, you need to create separate App ID and Provisioning Profile for it. It's simple as you already created it for main app.

I've attached screenshot of it to get more idea.
<img width="1256" alt="App ID" src="https://github.com/tedbarnett/MapWalk/assets/114083934/b11b0b2e-5d4b-4d7f-8919-80d8fe3f003b">
<img width="1299" alt="Provisioning Profile" src="https://github.com/tedbarnett/MapWalk/assets/114083934/eac9bce0-633c-437b-b828-99610da0057f">


After creating App ID and Provisioning Profile Action extension, you need to set it in Xcode like below screenshot.
As you already set main app's App ID and Provisioning Profile, now you need to only set it for Action extension.

Please check attached screenshots.

<img width="1034" alt="Xcode main app setup" src="https://github.com/tedbarnett/MapWalk/assets/114083934/6e90fd6e-defd-4131-bd38-823914119da7">


 
<img width="1062" alt="Xcode action extension setup" src="https://github.com/tedbarnett/MapWalk/assets/114083934/a306e1cb-df6d-4596-b70b-5e27a70e521e">


Please run Action extension first. You will be asked to choose app for Action extension, you can choose Message or Files, etc. to run it.
After that, you need to run main app.
Then you can test it. Please let me know if you need my help.

**Comment by tedbarnett** [comment]
I need to address those setup tasks you mentioned.

**Comment by bhautikpatoliya004** [comment]
Please let me know if you need my help.

**Comment by bhautikpatoliya004** [comment]
@tedbarnett This is already completed, but if you want to see Map Walk as a share recipient, you need to create a provisioning profile for this ACTION extension by following the steps above.

**Comment by tedbarnett** [comment]
Is that something you can set up for us or am I required?

**Comment by bhautikpatoliya004** [comment]
@tedbarnett Yes, I can setup for you but for that you need to invite me on your [Apple Developer account](https://idmsa.apple.com/IDMSWebAuth/signin?appIdKey=891bd3417a7776362562d2197f89480a8547b108fd934911bcbea0110d07f757&path=%2Faccount%2F&rv=1) and [App Store Connect](https://appstoreconnect.apple.com/login). After that I can create required provisioning profile and certificates/
If you have enterprise account then you can invite me to Apple developer account otherwise not.
Do you have enterprise apple developer account?

**Comment by tedbarnett** [comment]
Just set up a Barnett Labs organization that should work. Let me check...

**Comment by bhautikpatoliya004** [comment]
@tedbarnett I've created certificates, provisioning profile and added my UDID in the Device list in Apple developer account. Now, I successfully run app and also our app showing in share recipient list too.

You need to do some work from your side now.
First of all, this(00008130-00064D2A3420001C) UDID is already there in the apple developer account device list. If this is your device's UDID then you don't need to add it otherwise we need to add your iPhone UDID and need to update provisioning profile too.

I'have sent two provisioning profile in Upwork.
Now double click on "MapWalk_ActionExtension_dev_profile_1_Aug_2024.mobileprovision" and "MapWalk_dev_profile_1_Aug_2024.mobileprovision" file. It will install both App and it's extension profile in Xcode.

After installing profiles, you will see like below screenshot in Xcode.

![Uploading Screenshot 2024-08-12 at 2.41.13 PM.png…]
<img width="1022" alt="Screenshot 2024-08-12 at 2 41 49 PM" src="https://github.com/user-attachments/assets/4e2f3a89-fef0-4576-acf1-12e1f42d7f44">
()

If you trouble getting error in provisioning profile section of Xcode then 1-2 time close and open Xcode.
